src/test/ui/parser/keyword-const-as-identifier.stderr

```diff
@@ -1,13 +1,18 @@
-error: expected identifier, found keyword `const`
-  --> $DIR/keyword-const-as-identifier.rs:4:9
+error: expected `{`, found `=`
+  --> $DIR/keyword-const-as-identifier.rs:2:15
    |
 LL |     let const = "foo";
-   |         ^^^^^ expected identifier, found keyword
```

Comment:
This has regressed a bit now.
cc @oli-obk

Reply:
The thing is that this:

```rust
let const { "foo" } = "foo";
```

is a totally valid statement (although it ICEs; see #78174). Perhaps we could give the old warning if it's just `const` and then `=` immediately after?

Reply:
I see, but, what do you expect that code to do?.
I'd expect an error anyway something like `error[E0005]: refutable pattern in local binding: ``&_`` not covered`.

Reply:
It's valid _syntactically_, not _semantically_. (And either way it shouldn't ICE.)

Reply:
exactly, :+1: